homeassistant/components/elmax/binary_sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Elmax sensor platform."""
+from __future__ import annotations
+
+from elmax_api.model.panel import PanelStatus
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElmaxCoordinator
+from .common import ElmaxEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Elmax sensor platform."""
+    coordinator: ElmaxCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    known_devices = set()
+
+    def _discover_new_devices():
+        panel_status: PanelStatus = coordinator.data
+        # In case the panel is offline, its status will be None. In that case, simply do nothing
+        if panel_status is None:
+            return
```

Comment:
That's not really the case right? If the coordinator is unable to fetch new data isn't it just maintaining the old data until it's able to refresh?
Probably you should look at self.coordinator.last_update_success instead in that case but I think this can be removed instead as it will not do anything unless it finds new entities to handle.

Reply:
Please have a look at the comments previously discussed with @iMicknl [here](https://github.com/home-assistant/core/pull/64090#discussion_r784423267). This has already discussed: the coordinator will be able to fetch the data (the Elmax cloud will report the panel as OFFLINE), but there will be no data (as it's offline).